### ISSUE TITLE
data analyzer to automatically detect 'outliners'

### ISSUE THREAD

**Issue by wyli** [Project-MONAI/MONAI#5168]
as discussed, would be great to add some capability of detecting obvious outliners to `DataAnalyzer`
https://github.com/Project-MONAI/MONAI/blob/70c0443a703a8e5f9930d7b677d14425378b9226/monai/apps/auto3dseg/data_analyzer.py#L57-L64

cc @mingxin-zheng

**Comment by mingxin-zheng** [comment]
I have been thinking about this issue these days and would like to drop my thoughts here:

The usage can be:
- The outlier detector is compose which takes `Analyzer` transforms.
- It can be appended to transforms compose which includes `LoadImage` and other pre-processing transforms
- The structure of may be similar to [SegSummarizer](https://github.com/Project-MONAI/MONAI/blob/6e77d57938a4253353265616b54ab8be3f2d5ab3/monai/auto3dseg/seg_summarizer.py#L31)
- There is a `get_outlier` function which takes certain metric name to run analysis, and outputs outlier filenames, or slice# that help user to locate the outlier

User experience that may be interesting but also takes more hours to work on:
- Generate a standard HTML report. Each summary `Analyzer` generate standard texts and insert figures in the report. This may be useful for publication. For example, if a research paper is using private clinical dataset, they can put such report as supplemental materials. So the readers can understand the nature of the datasets.

To support the user experience and the functional requirements, we need to apply existing or create new outlier detection underlying methods. We have the following methods but can also add methods from other libraries such as [skicit-learn](https://scikit-learn.org/stable/modules/outlier_detection.html):

- Use the "percentile" definition from [existing operation](https://github.com/Project-MONAI/MONAI/blob/a209b06438343830e561a0afd41b1025516a8977/monai/auto3dseg/operations.py#L75). With the risk of over-simplification, for example, a sample will be considered as outlier, if a ceterin metric is below 0.5% or above 99.5%.
- Use the [histogram analyzer method](https://github.com/Project-MONAI/MONAI/blob/6e77d57938a4253353265616b54ab8be3f2d5ab3/monai/auto3dseg/analyzer.py#L814) contributed by @holgerroth to check the distribution. For example, if the distribution is assumed to be Gaussian, then identify the bins that is +/- 6 sigma away from the mean.

**Comment by wyli** [comment]
agreed, the detector API could be built to support customized detections such as simple rule-based classification (e.g. metadata field regular expression match), or based on some learned models? cc @diazandr3s who is recently working on dicom series selections

**Comment by diazandr3s** [comment]
This is a good discussion. I'm currently using the series selector from MONAI Deploy: https://github.com/Project-MONAI/monai-deploy-app-sdk/blob/main/monai/deploy/operators/dicom_series_selector_operator.py
It uses the DICOM tags and regular expressions. We could think of combining a basic heuristic algo along with computer vision models. Happy to brainstorm :)
